### PR TITLE
3.3/changelog

### PR DESCRIPTION
**Pull request by laclaro**
To do:
add explaining sentence concerning #1580 add correct release date.

### CODE REVIEW

File: Changelog [hunk 1]
```diff
@@ -1,17 +1,57 @@
-Version 3.3, 2020-XX-XX
+Version 3.3, *2020-XX-XX*
+
+  Features:
+    * New token type: WebAuthn/FIDO2 token (#1468)
+      * WebAuthn hardware tokens can now be used with privacyIDEA
+    * New token type: Indexed Secret. (#1986)
+      * A challenge-response token that asks the user for positions from a list of secrets.
+    * *OPEN New sophisticated logging facility (#1580)*
```
Comment (cornelinux) [inline Changelog]: As we talked I currently would remove this and add things that were commited after *now* add to the changelog in a 2nd round.

File: Changelog [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+      * *TEXT TEXT TEXT*
+
+  Fixes:
+    * Fixed unauthorized statistics view (#1238)
```
Comment (cornelinux) [inline Changelog]: In the past we used "Fix". Please do so, do not use "Fixed" and do not use "Fixes".

File: Changelog [hunk 3]
```diff
@@ -1,17 +1,57 @@
...
+    * Fixes a bug which caused an exception during PSKC key file container import (#1915)
+    * Fixed url_decode padding before it could cause any issues (#2000)
+    * Initialize rtype in user_object correctly (#2007)
+    * Fixed an inconsistency of start_tls with postgres SQL (#2025)
+    * Fix wrong type splitting of questionaire token (#2026)
+    * Fixed a bug which could cause missing audit entries when using the 
+      ContainerAudit module (#2029)
+    * Fixed a bug which prevented defining an SQL resolver without a password (#2030)
+    * Fixed missing "position" argument on event import with pi-manage (#2036)
+    * Fixed an issue that lead to failing tests in some cases (#2041)
```
Comment (cornelinux) [inline Changelog]: I am not sure if this is a change from 3.2 -> 3.3. Or if this simply something that happended during development of 3.3.
So then we should remove it from the changelog, since it is no change from 3.2->3.3

File: Changelog [hunk 4]
```diff
@@ -1,17 +1,57 @@
...
 
   Enhancements:
```
Comment (cornelinux) [inline Changelog]: Please put the Enhancements in front of the Fixes!

File: Changelog [hunk 5]
```diff
@@ -1,17 +1,57 @@
...
+         and default timestep for TOTP token can now be dependent on
+         admin user and for which user the admin does the enrollment
+        * Enrollment settings for push tokens can distinguish better
+         between admin users and user
+        * Random PIN settings can be user dependent
+      * Added a policy to define the allowed characters for PINs (#2049)
+    * WebUI
+      * Added the option to filter tokens by tokenrealm (#545)
+      * A QR code is displayed during enrollment of HOTP, TOTP and PUSH tokens which 
+        can be scanned with the authenticator app for quick key import (#1919).
```
Comment (cornelinux) [inline Changelog]: This is wrong. As we spoke we added a QR code to install an App before enrolling the token.
The QR code is supposed to lead the user to the play store or any other location, where he can install the app.

File: Changelog [hunk 6]
```diff
@@ -1,17 +1,57 @@
...
+      * Adding version hashes to WebUI components to avoid working with outdated data (#1871)
```
Comment (cornelinux) [inline Changelog]: Please change to "outdated templates". The data is always accurate.

File: Changelog (hunk 2 above)
Comment (laclaro) [reply Changelog]: ok

File: Changelog (hunk 4 above)
Comment (laclaro) [reply Changelog]: done

File: Changelog (hunk 5 above)
Comment (laclaro) [reply Changelog]: adjusted

File: Changelog (hunk 1 above)
Comment (laclaro) [reply Changelog]: ok. Removed open issues.

File: Changelog (hunk 3 above)
Comment (laclaro) [reply Changelog]: ok

File: Changelog (hunk 6 above)
Comment (laclaro) [reply Changelog]: that's right

File: Changelog [hunk 7]
```diff
@@ -1,17 +1,57 @@
...
+      * Updated bootstrap and JQuery to new versions (#830 #2061 #2064)
```
Comment (laclaro) [inline Changelog]: removed 830 and 2064 for now

File: Changelog [hunk 8]
```diff
@@ -1,17 +1,57 @@
-Version 3.3, 2020-XX-XX
+Version 3.3, *2020-XX-XX*
+
+  Features:
+    * New token type: WebAuthn/FIDO2 token (#1468)
+      * WebAuthn hardware tokens can now be used with privacyIDEA
+    * New token type: Indexed Secret. (#1986)
+      * A challenge-response token that asks the user for positions from a list of secrets.
```
Comment (laclaro) [inline Changelog]: removed also this one for now, issue 1986 still open

File: Changelog [hunk 9]
```diff
@@ -42,17 +24,30 @@ Version 3.3, *2020-XX-XX*
...
+    * Fix url_decode padding before it could cause any issues (#2000)
+    * Initialize rtype in user_object correctly (#2007)
+    * Fix an inconsistency of start_tls with postgres SQL (#2025)
+    * Fix wrong type splitting of questionnaire token (#2026)
+    * Fix a bug which could cause missing audit entries when using the
+      ContainerAudit module (#2029)
+    * Fix a bug which prevented defining an SQL resolver without a password (#2030)
+    * Fix missing "position" argument on event import with pi-manage (#2036)
+
 Version 3.2.2, 2020-01-17
```
Comment (laclaro) [inline Changelog]: Some editor was apparently too intelligent, removing blanks from the line endings...

**Comment by cornelinux** [comment]
@laclaro Please change it from Draft to Ready when you are done with my comments!
I want to merge it even if the changelog is not finalized!

File: Changelog [hunk 10]
```diff
@@ -164,10 +159,10 @@ Version 3.1, 2019-09-04
     * Do not increase fail counter when *checking* for an answered challenge (#1697)
     * Allow additional token specific checks when answering challenge response (#1695)
     * Endpoint GET /token/challenges also takes transaction_id (#1689)
-    * Push token can delay the response of /validate/check, so that there is no need
```
Comment (cornelinux) [inline Changelog]: @laclaro Why did you change all the old text? This is annoying in a PR and makes it hard to read and has nothing to do with the issue. Can you please check if you have some nasty autoformatting?
